test/test_ops.py

```diff
@@ -194,11 +197,13 @@ def check_variant_backward(self, input, forward_result, expected_grad, expected_
     @ops(op_db)
     def test_variant_consistency_eager(self, device, dtype, op):
         test_backward = op.supports_autograd and op.test_complex_grad or not dtype.is_complex
-        samples = op.sample_inputs(device, dtype, requires_grad=test_backward)
-        if len(samples) == 0:
+        try:
+            samples = iter(op.sample_inputs(device, dtype, requires_grad=test_backward))
+            first_sample = next(samples)
```

Comment:
This is a little awkward but luckily there aren't many tests in test_ops.py so it's not a huge burden to impose.

Reply:
That said... what would you think about adding a op.single_sample() helper?

This helper would do two things:

1) it could be overridden in case there is some especially representative single sample the op wants to use when only testing one sample
2) if unspecified, it just acquires the first sample from sample_inputs() or returns None

Follow-up question: do any ops not have sample_inputs()? I think there was some idea this might happen but in practice all ops probably have to have sample_inputs() -- what would we do with them if they didn't? If so, we can simply avoid these checks for an op having no sample inputs. I still think supporting a single_sample() helper would be useful, though.

Reply:
Could you please elaborate where the burden might come from?

Reply:
I just think it would be more straightforward for the test to directly express what it wants: to acquire a single sample.

Reply:
I do not know about the methods that do not accept samplers to be honest, and I see no checks for that in the `test_ops.py`.
I do not mind having an additional `op.single_sample` but then checking and skipping on None is not that fundamentally different from what this PR is doing.  Wouldn't it actually confuse the user even more? The `sample_inputs` is already doing the job. Maybe we could just make a protected helper method inside the `TestOpInfo` test case, without adding an additional attribute to OpInfo?

Reply:
Maybe. Why don't we just simplify by always assuming that there is at least one SampleInput so we don't have to catch the exception? We're already updating the sample input documentation, so this PR can just further extend it to clarify there should always be at least one sample input returned.

Reply:
Sure, if this is the case, then I am totally fine with this. It would definitely make things easier, given that there are no checks for `sample_inputs` even though the default value for it is `None`.